File: study-design-patterns/java/factory-pattern/src/product/DbHandler.java
package product;

public class DbHandler extends IOHandler{
    @Override
    public void add(String id, String name) {
        System.out.print("DbHandler add method call!\n");
    }

    @Override
    public void remove(String id) {
        System.out.print("DbHandler remove method call!\n");
    }

    @Override
    public void update(String id, String name) {
        System.out.print("DbHandler update method call!\n");
    }

    @Override
    public void query(String id) {
        System.out.print("DbHandler query method call!\n");
    }
}

File: study-design-patterns/ts/factory-pattern/product/DbHandler.ts
import IOHandler from "./IOHandler";

export default class DbHandler extends IOHandler {
  add(): void {
    console.log('DbHandler add method call!')
  }

  query(): void {
    console.log('DbHandler query method call!')
  }

  remove(): void {
    console.log('DbHandler remove method call!')
  }

  update(): void {
    console.log('DbHandler update method call!')
  }
}